enumerables.rb

```diff
@@ -0,0 +1,332 @@
+# rubocop:disable Metrics/CyclomaticComplexity, Metrics/MethodLength, Metrics/PerceivedComplexity, Lint/Void
+p '-----------|||||--------'
+p '----------{(o o)}--------'
+p '--------ooO-(_)-Ooo------'
+# Initial comment
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    array = to_a
+    size.times { |index| yield(array[index]) }
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    array = to_a
+    size.times { |index| yield(array[index], index) }
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    array = to_a
+    arraya = []
+    array.length.times do |a|
+      array[a]
+      arraya << array[a] if yield (array[a])
+    end
+    arraya
+  end
+
+  def my_all?(*arg)
+    return 'error arguments' if arg.length > 1
+
+    if block_given?
+      array = to_a
+      cont = 0
+      array.length.times do |a|
+        array[a]
+        if yield (array[a])
+          cont += 1
+          return true if cont == array.length
+
+          next
+        end
+        return false
+      end
+      true elsif arg.empty?
+    elsif arg[0].class == Class
+      array3 = to_a
+      cont2 = 0
+      array3.length.times do |b|
+        if array3[b].is_a?(arg[0])
+          cont2 += 1
+          return true if cont2 == array3.length
+
+          next
+        end
+        return false
+      end
+    elsif arg[0].class == Regexp
+      array3 = to_a
+      cont2 = 0
+      array3.length.times do |b|
+        if arg[0].match(array3[b])
+          cont2 += 1
+          return true if cont2 == array3.length
+
+          next
+        end
+        return false
+      end
+    end
+  end
+end
+
+module Enumerable
+  def my_any?(*arg)
```

Comment:
Should returns false if the block ever returns a value false or nil.
![Screen Shot 2020-07-21 at 7 33 07 PM](https://user-images.githubusercontent.com/52430951/88120642-069afc80-cb89-11ea-8afe-61a4cedaf0b1.png)
****
When no block or argument is given returns true if at least one of the collection is not false or nil.
![Screen Shot 2020-07-21 at 7 34 01 PM](https://user-images.githubusercontent.com/52430951/88120692-26322500-cb89-11ea-984f-5d1ffaa785ac.png)
****
When no block or argument is given returns false if at least one of the collection is not true.
![Screen Shot 2020-07-21 at 7 35 26 PM](https://user-images.githubusercontent.com/52430951/88120768-58438700-cb89-11ea-9c41-0e7a8d4ba72e.png)
****
When a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern.
![Screen Shot 2020-07-21 at 7 37 48 PM](https://user-images.githubusercontent.com/52430951/88120900-ad7f9880-cb89-11ea-99ab-a72dc965d301.png)
****
When a pattern other than Regex or a Class is given returns true if any of the collection matches the pattern
![Screen Shot 2020-07-21 at 7 38 38 PM](https://user-images.githubusercontent.com/52430951/88120926-cab46700-cb89-11ea-891b-5497dfdb497c.png)